src/overview/components/DateRangeSelection.jsx

```diff
@@ -1,46 +1,125 @@
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='after..'
-            isClearable
-            selected={startDate && moment(startDate)}
-            selectsStart
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onStartDateChange(
-                date ? date.valueOf() : undefined,
-            )}
-        />
-        <DatePicker
-            className={styles.datePicker}
-            dateFormat='DD-MM-YYYY'
-            placeholderText='before..'
-            isClearable
-            selected={endDate && moment(endDate)}
-            selectsEnd
-            startDate={moment(startDate || 0)}
-            endDate={moment(endDate)}
-            maxDate={moment()}
-            onChange={date => onEndDateChange(
-                date ? date.endOf('day').valueOf() : undefined,
-            )}
-        />
-    </div>
-)
+class DateRangeSelection extends Component {
+    constructor() {
+        super()
+
+        this.state = {
+            startDateText: null,
+            endDateText: null,
+        }
+
+        this.handleKeydown = this.handleKeydown.bind(this)
+        this.submitDateChange = this.submitDateChange.bind(this)
+    }
+
+    componentDidMount() {
+        this.refs.startDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'startDate'))
+        this.refs.endDateDatePicker.refs.input.refs.input.addEventListener('keydown', e => this.handleKeydown(e, 'endDate'))
+    }
+
+    handleKeydown(event, type) {
+        if (event.key === 'Enter') {
+            event.stopImmediatePropagation()
+            this.submitDateChange(type)
+        }
+    }
+
+    submitDateChange(type) {
+        if (type === 'startDate') {
+            const date = moment(this.state.startDateText, 'DD-MM-YYYY', true)
+            const nlpDate = chrono.parseDate(this.state.startDateText)
+
+            const dateToChange = date.isValid()
+                ? date.valueOf()
+                : nlpDate && nlpDate.getTime()
+
+            if (dateToChange !== this.props.startDate) this.props.onStartDateChange(dateToChange)
+        } else if (type === 'endDate') {
+            const date = moment(this.state.endDateText, 'DD-MM-YYYY', true)
+            const nlpDate = chrono.parseDate(this.state.endDateText)
+
+            const dateToChange = date.isValid()
+                ? date.valueOf()
+                : nlpDate && nlpDate.getTime()
+
+            if (dateToChange !== this.props.endDate) this.props.onEndDateChange(dateToChange)
+        }
+    }
+
+    render() {
+        const {
+            startDate, endDate,
+            onStartDateChange, onEndDateChange,
+        } = this.props
+
+        return (
+            <div className={styles.dateRangeSelection}>
+                <DatePicker
+                    ref='startDateDatePicker'
```

Comment:
Make this a simple arrow fn like `ref={ref => { this.blah = ref }}` to avoid the whole string ref pattern, then can update the stuff in `componentDidMount()` from `this.refs.blah` to `this.blah` to get the explicitly assigned ref.